packages/core/src/connect/message-signing/verify.ts

```diff
@@ -129,11 +131,47 @@ export const verifyStructuredMessageSignature = (options: {
 
   const hashBytes = makeStructuredDataHash(domain, message);
 
-  return verifyMessageSignature({
+  return _verifyMessageSignature({
     message: hashBytes,
     signature: options.signature,
     publicKey: options.publicKey,
     stxAddress: options.stxAddress,
     mode: options.mode,
+    prefix: options.prefix,
   });
 };
+
+export const verifyMessageSignature = (options: {
+  // string = message, bytes = hash
+  message: string | Uint8Array;
+  signature: string;
+  publicKey?: string;
+  stxAddress?: string;
+  mode?: 'vrs' | 'rsv';
+}) => {
+  const isValid = _verifyMessageSignature(options);
+  return isValid
+    ? true
+    : _verifyMessageSignature({
+        ...options,
+        prefix: LEGACY_CHAIN_PREFIX_BYTES,
+      });
+};
+
+export const verifyStructuredMessageSignature = (options: {
+  message: StructuredSignatureRequestOptions['message'];
+  domain: StructuredSignatureRequestOptions['domain'];
+  signature: string;
+  publicKey?: string;
+  stxAddress?: string;
+  mode?: 'vrs' | 'rsv';
+}) => {
+  const isValid = _verifyStructuredMessageSignature(options);
+  return (
+    isValid ??
```

Comment:
I haven‘t tested this but `??` might be problematic here: nullish vs falsy ([nullish coalescing](https://www.joshwcomeau.com/operator-lookup?match=nullish-coalescing)) [replit](https://replit.com/@janniks1/TemptingNecessaryNumerators#index.ts)